### PR TITLE
chore(kuma-cp): improve docs around KDS client-id/peerID

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/kds/mux/server.go [hunk 1]
```diff
@@ -174,11 +174,11 @@ func (s *server) Start(stop <-chan struct{}) error {
 
 // StreamMessage handle Mux messages for KDS V1. It's not used in KDS V2
 func (s *server) StreamMessage(stream mesh_proto.MultiplexService_StreamMessageServer) error {
-	clientID, err := util.ClientIDFromIncomingCtx(stream.Context())
+	zoneID, err := util.ClientIDFromIncomingCtx(stream.Context())
 	if err != nil {
 		return err
 	}
-	log := muxServerLog.WithValues("client-id", clientID)
+	log := muxServerLog.WithValues("client-id", zoneID)
```
Comment (bartsmykla) [inline pkg/kds/mux/server.go:181]: Should we use `peer-id` here like in [`pkg/kds/global/components.go`](https://github.com/kumahq/kuma/pull/8897/files#diff-4bfb1199f26a2d0217a66d2f0262151b44248e329a0d7cabd15acb2a1b10e557R120)?

File: pkg/kds/util/client_id.go [hunk 2]
```diff
@@ -2,22 +2,27 @@ package util
...
 
-func MetadataFromIncomingCtx(ctx context.Context, key string) (string, error) {
+func metadataFromIncomingCtx(ctx context.Context, key string) (string, error) {
 	md, ok := metadata.FromIncomingContext(ctx)
 	if !ok {
 		return "", errors.New("metadata is not provided")
 	}
 	if len(md[key]) == 0 {
-		return "", errors.New("'client-id' is not present in metadata")
+		return "", errors.New(fmt.Sprintf("'%s' is not present in metadata", clientIDKey))
```
Comment (bartsmykla) [inline pkg/kds/util/client_id.go:25]:
```suggestion
		return "", errors.New(fmt.Sprintf("%q is not present in metadata", key))
```

**Review by bartsmykla — CHANGES_REQUESTED** [review]
I have one question and there is one small mistake in `metadataFromIncomingCtx`

File: pkg/kds/mux/server.go (hunk 1 above)
Comment (michaelbeaumont) [reply pkg/kds/mux/server.go:181]: I didn't want to change any logs or anything in this PR, perhaps in https://github.com/kumahq/kuma/pull/8898

Comment (bartsmykla) [reply pkg/kds/mux/server.go:181]: got it

**Pull request merged**